paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalogFactory.java

```diff
@@ -69,7 +69,7 @@ public Catalog create(FileIO fileIO, Path warehouse, CatalogContext context) {
         // always using user-set parameters overwrite hive-site.xml parameters
         context.options().toMap().forEach(hiveConf::set);
         hiveConf.set(HiveConf.ConfVars.METASTOREURIS.varname, uri);
-        hiveConf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, warehouse.toUri().toString());
+        hiveConf.set(CatalogOptions.WAREHOUSE.key(), warehouse.toUri().toString());
```

Comment:
Can we delete this line?
We can just pass warehouse to `HiveCatalog`.
And just pass warehouse to `LoctionHelper`.

Reply:
It's better for passing warehouse to HiveCatalog for a better readable indeed.